csp_billing_adapter/exceptions.py

```diff
@@ -49,3 +49,7 @@ class FailedToSaveCSPConfigError(CSPBillingAdapterException):
 
 class FailedToSaveCacheError(CSPBillingAdapterException):
     """Exception raised when cache save/update fails."""
+
+
+class MetadataCollectorError(CSPBillingAdapterException):
```

Comment:
Wondering if `MetadataRetrievalError` might be a better name?

And perhaps should it have a CSP prefix, given that it is an exception specific to the CSP integration plugins?